### ISSUE TITLE
Confusing and misleading version string requirements: The VS Marketplace doesn't support prerelease versions

### ISSUE THREAD

**Issue by ssbarnea** [microsoft/vscode-vsce#806]
The error message [The VS Marketplace doesn't support prerelease versions](https://github.com/microsoft/vscode-vsce/blob/main/src/publish.ts#L145) contradicts the official documentation from https://code.visualstudio.com/api/working-with-extensions/publishing-extension#autoincrementing-the-extension-version which states that version numbers follow https://semver.org/ 

While SemVer states that a pre-release version looks like `1.2.3-4` that is not allowed for publishing and that is recent regression, as previously it did allow them.

Basically it seems that is now impossible to make use of any pre-release versioning as the only one allowed are `X.Y.Z` which based on SemVer are always release versions.

**Comment by isidorn** [comment]
Closing as dup of https://github.com/microsoft/vsmarketplace/issues/50

From your description you are saying this is a recent regression but I believe this never worked.
If this worked do let us know when exactly it did.
Thanks!
